apps/src/blockly/customBlocks/googleBlockly/proceduresBlocks.js

```diff
@@ -138,6 +133,21 @@ GoogleBlockly.Extensions.register('procedures_block_frame', function () {
   }
 });
 
+// Override the destroy function to not destroy the procedure. We need to do this
+// so that when we clear the modal function editor we don't remove the procedure
+// from the procedure map.
+GoogleBlockly.Extensions.register('modal_procedures_no_destroy', function () {
+  const mixin = {
+    destroy: function () {
+      // no-op
+      // this overrides the destroy hook registered
+      // in the procedure_def_get_def_mixin
+    },
+  };
+  // We can't register this as a mixin since we're overwriting existing methods
+  Object.assign(this, mixin);
```

Comment:
The workaround makes sense. Did Maribeth give us any guidance here?

Reply:
Yep, in fact I copied this from her proof of concept.